packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1129,6 +1129,16 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   // [assembleSemanticsNode] invocations.
   Queue<SemanticsNode>? _cachedChildNodes;
 
+  /// Returns a list of rects that bound the given selection.
+  ///
+  /// See [TextPainter.getBoxesForSelection] for more details.
+  List<Rect> getBoxesForSelection(TextSelection selection) {
```

Comment:
Would it be any cleaner if this method were added to [TextLayoutMetrics](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/text_layout_metrics.dart)?

The goal of that class is to simplify things by providing a read-only interface instead of the full RenderEditable.  You probably have to use RenderEditable anyway though, so maybe it's not worth it here.  Just wanted to bring it up.